keras_nlp/models/bart/bart_seq_2_seq_lm.py

```diff
@@ -193,7 +193,7 @@ def __init__(
         # Use token embedding weights to project from the token representation
         # to vocabulary logits.
         outputs = tf.matmul(
-            x,
```

Comment:
I think we actually want this the other way. We should cast the variable to the compute dtype (the lower precision type when using mixed), before multiplying with `x`. So `tf.cast(backbone.token_embedding.embeddings, x.dtype)`.

Can you check if that fixes things as well?

Reply:
Hmmm, why does GPT-2 output `tf.float32` instead of `tf.float16`, then?

Reply:
In what context does it? There might be some casting going on to the outputs.

My understanding is that we should be casting our variables to the `compute_dtype`. Take a look at
https://github.com/keras-team/keras/blob/v2.12.0/keras/engine/base_layer.py#L2216-L2235
and https://github.com/keras-team/keras/blob/v2.12.0/keras/mixed_precision/autocast_variable.py

But maybe I am missing something!

Reply:
GPT2 backbone outputs tf.float32. BART backbone outputs tf.float16. Weird why that's happening.

https://colab.research.google.com/drive/18AbKIwbUAtJySgAYWXa0ggvRbp5TriEa?usp=sharing

Reply:
This actually may no longer be necessary with the port. We had to wrap things in a Layer to avoid some errors with keras-core, which might mean that variables are now autocasting automatically (I think it should).

Can you try again on the latest from master?